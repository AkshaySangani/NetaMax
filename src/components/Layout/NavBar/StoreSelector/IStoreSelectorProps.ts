import { FlexProps } from "@chakra-ui/react";

export interface IStoreSelectorProps extends FlexProps {
  /**
   * The store selector.
   * @type {string}
   */
  storeName?: string;

  /**
   * Indicates if the container has padding
   * @type {boolean}
   **/
  hasPadding?: boolean;
}
