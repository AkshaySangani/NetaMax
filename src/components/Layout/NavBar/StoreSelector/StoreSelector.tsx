import React, { ReactElement } from "react";

import UbicationIcon from "styled/icons/Ubication";

import { Box, Flex, Text } from "@chakra-ui/react";

import { IStoreSelectorProps } from "./IStoreSelectorProps";

/**
 * The store selector component
 * @param {IStoreSelectorProps} props the props for the store selector
 * @returns {ReactElement} the store selector component.
 */
export const StoreSelector = (props: IStoreSelectorProps): ReactElement => {
  const { storeName, hasPadding, ...rest } = props;

  return (
    <Flex alignItems="center" pt={hasPadding ? "5px" : "0px"} justifyContent="flex-start" {...rest}>
      <Box padding="10px">
        <UbicationIcon />
      </Box>
      <Text
        variant="unstyled"
        fontWeight="700"
        color="white"
        fontSize={{ base: "16px", sm: "14px", md: "16px", lg: "18px" }}
      >
        Recoge en: {storeName}
      </Text>
    </Flex>
  );
};
