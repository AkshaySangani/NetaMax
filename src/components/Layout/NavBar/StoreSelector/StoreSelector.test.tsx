import { render, cleanup } from "@testing-library/react";
import { StoreSelector } from "./StoreSelector";
import { IStoreSelectorProps } from "./IStoreSelectorProps";

describe("StoreSelector Component", () => {
  let mock: IStoreSelectorProps;

  beforeEach(() => {
    mock = {
      storeName: "Store 1",
      hasPadding: true,
    };
  });

  afterEach(() => {
    cleanup();
  });

  test("StoreSelector should be render correctly", () => {
    const { container } = render(<StoreSelector {...mock} />);
    expect(container).toMatchSnapshot();
  });

  test("StoreSelector should be also render correctly with false hasPadding flag", () => {
    mock = {
      storeName: "Store 1",
      hasPadding: false,
    };
    const { container } = render(<StoreSelector {...mock} />);
    expect(container).toMatchSnapshot();
  });
});
