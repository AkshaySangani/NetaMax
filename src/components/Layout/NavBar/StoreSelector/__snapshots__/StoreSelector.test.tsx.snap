// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreSelector Component StoreSelector should be also render correctly with false hasPadding flag 1`] = `
<div>
  <div
    class="css-d18oyi"
  >
    <div
      class="css-nk32ej"
    >
      <svg
        fill="none"
        height="25"
        viewBox="0 0 20 25"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M4.45804 10.6666C4.45804 7.60601 6.93913 5.12492 9.9997 5.12492C13.0603 5.12492 15.5414 7.60601 15.5414 10.6666C15.5414 13.7272 13.0603 16.2083 9.9997 16.2083C6.93913 16.2083 4.45804 13.7272 4.45804 10.6666ZM9.9997 6.87492C7.90562 6.87492 6.20804 8.57251 6.20804 10.6666C6.20804 12.7607 7.90562 14.4583 9.9997 14.4583C12.0938 14.4583 13.7914 12.7607 13.7914 10.6666C13.7914 8.57251 12.0938 6.87492 9.9997 6.87492Z"
          fill="white"
          fill-rule="evenodd"
        />
        <path
          clip-rule="evenodd"
          d="M0.111461 9.33291C0.524843 4.31787 4.71569 0.458252 9.74774 0.458252H10.2517C15.2837 0.458252 19.4746 4.31787 19.8879 9.33291C20.11 12.0269 19.2778 14.702 17.5667 16.7946L11.9748 23.6333C10.954 24.8818 9.0454 24.8818 8.02457 23.6333L2.43266 16.7946C0.72156 14.702 -0.110601 12.0269 0.111461 9.33291ZM9.74774 2.20825C5.62645 2.20825 2.19411 5.36931 1.85555 9.47668C1.67073 11.7188 2.36332 13.9452 3.78742 15.6868L9.37932 22.5256C9.69996 22.9177 10.2994 22.9177 10.6201 22.5256L16.212 15.6868C17.6361 13.9452 18.3287 11.7188 18.1439 9.47668C17.8053 5.36931 14.373 2.20825 10.2517 2.20825H9.74774Z"
          fill="white"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <p
      class="chakra-text css-njr5bn"
    >
      Recoge en: 
      Store 1
    </p>
  </div>
</div>
`;

exports[`StoreSelector Component StoreSelector should be render correctly 1`] = `
<div>
  <div
    class="css-d9urq1"
  >
    <div
      class="css-nk32ej"
    >
      <svg
        fill="none"
        height="25"
        viewBox="0 0 20 25"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M4.45804 10.6666C4.45804 7.60601 6.93913 5.12492 9.9997 5.12492C13.0603 5.12492 15.5414 7.60601 15.5414 10.6666C15.5414 13.7272 13.0603 16.2083 9.9997 16.2083C6.93913 16.2083 4.45804 13.7272 4.45804 10.6666ZM9.9997 6.87492C7.90562 6.87492 6.20804 8.57251 6.20804 10.6666C6.20804 12.7607 7.90562 14.4583 9.9997 14.4583C12.0938 14.4583 13.7914 12.7607 13.7914 10.6666C13.7914 8.57251 12.0938 6.87492 9.9997 6.87492Z"
          fill="white"
          fill-rule="evenodd"
        />
        <path
          clip-rule="evenodd"
          d="M0.111461 9.33291C0.524843 4.31787 4.71569 0.458252 9.74774 0.458252H10.2517C15.2837 0.458252 19.4746 4.31787 19.8879 9.33291C20.11 12.0269 19.2778 14.702 17.5667 16.7946L11.9748 23.6333C10.954 24.8818 9.0454 24.8818 8.02457 23.6333L2.43266 16.7946C0.72156 14.702 -0.110601 12.0269 0.111461 9.33291ZM9.74774 2.20825C5.62645 2.20825 2.19411 5.36931 1.85555 9.47668C1.67073 11.7188 2.36332 13.9452 3.78742 15.6868L9.37932 22.5256C9.69996 22.9177 10.2994 22.9177 10.6201 22.5256L16.212 15.6868C17.6361 13.9452 18.3287 11.7188 18.1439 9.47668C17.8053 5.36931 14.373 2.20825 10.2517 2.20825H9.74774Z"
          fill="white"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <p
      class="chakra-text css-njr5bn"
    >
      Recoge en: 
      Store 1
    </p>
  </div>
</div>
`;
